cps/web.py

```diff
@@ -2115,6 +2116,13 @@ def upload():
     db.session.connection().connection.connection.create_function('uuid4', 0, lambda: str(uuid4()))
     if request.method == 'POST' and 'btn-upload' in request.files:
         file = request.files['btn-upload']
+        if not ('.' in file.filename and file.filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS):
+            flash(
+                _('File extension "%s" is not allowed to be uploaded to this server' % 
+                file.filename.rsplit('.', 1)[1].lower()),
```

Comment:
This will cause a `list index out of range` error when user uploads a file without any dot, i.e without an extension